pytorch_lightning/plugins/training_type/single_device.py

```diff
@@ -78,3 +80,26 @@ def barrier(self, *args, **kwargs) -> None:
 
     def broadcast(self, obj: object, src: int = 0) -> object:
         return obj
+
+    def teardown(self) -> None:
+        """
+        This method is called to teardown the training process.
+        It is the right place to release memory and free other ressources.
+
+        By default, we teardown in the following way:
+            1. if training is on gpu, we move lightning module to CPU and clean up
+            cuda memory;
+            2. if training is on tpu, we clean up os environment debug flag
+        """
+        if self.on_gpu:
+            # GPU teardown
+            self.lightning_module.cpu()
+            # clean up memory
+            with torch.cuda.device(self.root_device):
+                torch.cuda.empty_cache()
+            return
+        if self.on_tpu:
```

Comment:
Maybe we could the tpu logic to the `single_tpu` training plugin?

Reply:
@kaushikb11 n00b question: why is single_tpu its own plugin vs using the single-device plugin?

Reply:
We have it decoupled. We could find a way to have it as one, but it would get messy with conditions and error-prone and to write tpu specific code.